core/templates/services/image-local-storage.service.ts

```diff
@@ -110,6 +110,43 @@ export class ImageLocalStorageService {
     this.storedImageFilenames.length = 0;
     this.thumbnailBgColor = null;
   }
+
+  async getFilenameToBase64MappingAsync(
```

Comment:
Where are the frontend tests for this?

Reply:
Added a test!

Reply:
Please add a test for the following O/P

1. `reject(new Error('No valid prefix found in data url'));`
2. `throw new Error('No image data found');`
3. when `imagesData.length == 0`

Reply:
It looks like these cases are explicitly covered in `translate-text-backend-api.service.ts`. Added tests anyway!